internal/outofband/action_handlers.go

```diff
@@ -734,39 +763,35 @@ func (h *actionHandler) resetBMC(a sw.StateSwitch, c sw.TransitionArgs) error {
 		return err
 	}
 
-	// proceed with reset only if these flags are set
-	if !action.BMCPowerCycleRequired && !tctx.Task.Parameters.ResetBMCBeforeInstall {
-		return nil
-	}
-
 	tctx.Logger.WithFields(
 		logrus.Fields{
-			"component":                             action.Firmware.Component,
-			"bmc":                                   tctx.Asset.BmcAddress,
-			"task.Parameters.ResetBMCBeforeInstall": tctx.Task.Parameters.ResetBMCBeforeInstall,
-			"action.BMCPowerCycleRequired":          action.BMCPowerCycleRequired,
-		}).Info("resetting BMC")
+			"component": action.Firmware.Component,
+			"bmc":       tctx.Asset.BmcAddress,
+		}).Info("resetting BMC, delay introduced: " + delayBMCReset.String())
 
-	if !tctx.Dryrun {
-		if err := tctx.DeviceQueryor.ResetBMC(tctx.Ctx); err != nil {
-			return err
-		}
+	err = h.powerCycleBMC(tctx)
```

Comment:
This isn't conditional on `tctx.DryRun`? Should it be?

_Edit:_ NVM, it's in the `powerCycleBMC` method